src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -0,0 +1,257 @@
+        }
+
+        private async Task<bool> RunOperationProcessorsAsync(SwaggerDocument document, ApiDescription apiDescription, Type controllerType, MethodInfo methodInfo, SwaggerOperationDescription operationDescription, List<SwaggerOperationDescription> allOperations, SwaggerGenerator swaggerGenerator, SwaggerSchemaResolver schemaResolver)
+        {
+            // 1. Run from settings
+            var operationProcessorContext = new AspNetCoreOperationProcessorContext(document, operationDescription, controllerType, methodInfo, swaggerGenerator, _schemaGenerator, schemaResolver, allOperations)
+            {
+                ApiDescription = apiDescription,
+            };
+            foreach (var operationProcessor in Settings.OperationProcessors)
+            {
+                if (await operationProcessor.ProcessAsync(operationProcessorContext).ConfigureAwait(false) == false)
+                    return false;
+            }
+
+            // 2. Run from class attributes
+            var operationProcessorAttribute = methodInfo.DeclaringType.GetTypeInfo()
+                .GetCustomAttributes()
+            // 3. Run from method attributes
+                .Concat(methodInfo.GetCustomAttributes())
+                .Where(a => a.GetType().Name == "SwaggerOperationProcessorAttribute");
+
+            foreach (dynamic attribute in operationProcessorAttribute)
+            {
+                var operationProcessor = ReflectionExtensions.HasProperty(attribute, "Parameters") ?
+                    Activator.CreateInstance(attribute.Type, attribute.Parameters) :
+                    Activator.CreateInstance(attribute.Type);
+
+                if (operationProcessor.Process(methodInfo, operationDescription, swaggerGenerator, allOperations) == false)
+                    return false;
+            }
+
+            return true;
+        }
+
+        private string GetOperationId(SwaggerDocument document, string controllerName, MethodInfo method)
+        {
+            string operationId;
+
+            dynamic swaggerOperationAttribute = method.GetCustomAttributes().FirstOrDefault(a => a.GetType().Name == "SwaggerOperationAttribute");
+            if (swaggerOperationAttribute != null && !string.IsNullOrEmpty(swaggerOperationAttribute.OperationId))
+                operationId = swaggerOperationAttribute.OperationId;
+            else
+            {
+                if (controllerName.EndsWith("Controller"))
+                    controllerName = controllerName.Substring(0, controllerName.Length - 10);
+
+                operationId = controllerName + "_" + GetActionName(method);
+            }
+
+            var number = 1;
+            while (document.Operations.Any(o => o.Operation.OperationId == operationId + (number > 1 ? "_" + number : string.Empty)))
+                number++;
+
+            return operationId + (number > 1 ? number.ToString() : string.Empty);
+        }
+
+        private string GetActionName(MethodInfo method)
```

Comment:
Should this use the actiondescriptor?

Reply:
The operation id should be in the form “controller_operation” to ensure correct code generation (multiple client classes per controller and same operation/method names)